File: lib/domain/shared/ports/fetch_weather_data_enqueue_port.rb
# frozen_string_literal: true

module Domain
  module Shared
    module Ports
      module FetchWeatherDataEnqueuePort
        # @param blocks [Array<Hash>] :start_date, :end_date
        def enqueue_farm_weather_fetch(farm_id:, latitude:, longitude:, blocks:)
          raise NotImplementedError, "#{self.class}#enqueue_farm_weather_fetch"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/shared/ports/fetch_weather_data_enqueue_port.rs
use crate::shared::dtos::WeatherFetchDateBlock;

/// Ruby: `Domain::Shared::Ports::FetchWeatherDataEnqueuePort`
pub trait FetchWeatherDataEnqueuePort: Send + Sync {
    fn enqueue_farm_weather_fetch(
        &self,
        farm_id: i64,
        latitude: f64,
        longitude: f64,
        blocks: &[WeatherFetchDateBlock],
    );
}
